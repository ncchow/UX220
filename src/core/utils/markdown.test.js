/**
 * @file markdown
 * @author Cuttle Cong
 * @date 2018/2/24
 * @description
 * @jest-environment node
 */
import mark from './markdown'
function j(arr = []) {
  return arr.join('\n')
}

describe('markdown', function () {
  test('should markdown works about normal case', async () => {
    const html = await mark(j([
      '# Heading 1',
      '## Heading 2',
      'normal **bold** text'
    ]))
    expect(
      html
    ).toMatchSnapshot()
  })

  test('should markdown works about gemoji', async () => {
    expect(
      await mark(j([
        ':smile: ðŸ‘¼',
        ':+1: :dash:',
      ]))
    ).toMatchSnapshot()

    expect(
      await mark(j([
        ':smile: ðŸ‘¼  ', // should insert new line
        ':+1: :dash:',
      ]))
    ).toMatchSnapshot()
  })

  test('should works about task lists', async () => {
    expect(
      await mark(j([
        '[x] not ok',
        '- [x] done',
        '- [] NOT matching',
        '- [ ] todo',
      ]))
    ).toMatchSnapshot()
  })

  test('should works about yaml preamble', async () => {
    expect(
      await mark(j([
        '---',
        'title: abc',
        'tags: [a, b, c]',
        '---'
      ]))
    ).toMatchSnapshot()
  })

  test('should works about replace link', async () => {
    expect(
      await mark(j([
        '[abc](./readme.md)\\', // break line
        '<a href="./readme.md">exact link</a>'
      ]), {
        markdownItOptions: {
          replaceLink(link, env) {
            return link + "?c=123"
          }
        }
      })
    ).toMatchSnapshot()
  })

  test('should works about replace link with fromUrl', async () => {
    expect(
      await mark(j([
        '[abc](./readme.md)\\', // break line
      ]), {
        fromUrl: 'http://github.com/telescopejs/telescope/',
        markdownItOptions: {
        }
      })
    ).toMatchSnapshot()
  })
})
