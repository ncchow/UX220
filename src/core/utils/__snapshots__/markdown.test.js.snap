// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markdown should markdown works about gemoji 1`] = `
"<p>😄 👼
👍 💨</p>
"
`;

exports[`markdown should markdown works about gemoji 2`] = `
"<p>😄 👼<br>
👍 💨</p>
"
`;

exports[`markdown should markdown works about normal case 1`] = `
"<h1><a id=\\"user-content-heading-1\\" class=\\"anchor\\" href=\\"#heading-1\\" aria-hidden=\\"true\\"><svg aria-hidden=\\"true\\" class=\\"octicon octicon-link\\" height=\\"16\\" version=\\"1.1\\" viewbox=\\"0 0 16 16\\" width=\\"16\\"><path d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg></a>Heading 1</h1>
<h2><a id=\\"user-content-heading-2\\" class=\\"anchor\\" href=\\"#heading-2\\" aria-hidden=\\"true\\"><svg aria-hidden=\\"true\\" class=\\"octicon octicon-link\\" height=\\"16\\" version=\\"1.1\\" viewbox=\\"0 0 16 16\\" width=\\"16\\"><path d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg></a>Heading 2</h2>
<p>normal <strong>bold</strong> text</p>
"
`;

exports[`markdown should works about replace link 1`] = `
"<p><a href=\\"readme.md\\">abc</a><br>
<a href=\\"./readme.md\\">exact link</a></p>
"
`;

exports[`markdown should works about replace link with fromUrl 1`] = `
"<p><a href=\\"http://github.com/telescopejs/telescope/readme.md\\">abc</a>\\\\</p>
"
`;

exports[`markdown should works about task lists 1`] = `
"<p>[x] not ok</p>
<ul class=\\"contains-task-list\\">
<li class=\\"task-list-item\\"><input class=\\"task-list-item-checkbox\\" checked=\\"\\" disabled=\\"\\" type=\\"checkbox\\"> done</li>
<li>[] NOT matching</li>
<li class=\\"task-list-item\\"><input class=\\"task-list-item-checkbox\\" disabled=\\"\\" type=\\"checkbox\\"> todo</li>
</ul>
"
`;

exports[`markdown should works about yaml preamble 1`] = `
"<table>
<thead>
<tr>
<th>title</th>
<th>tags</th>
</tr>
</thead>
<tbody>
<tr>
<td>abc</td>
<td>a,b,c</td>
</tr>
</tbody>
</table>
"
`;
